membership_extension/models/account_invoice.py

```diff
@@ -22,39 +22,24 @@ def action_invoice_draft(self):
     def action_cancel(self):
         """Cancel membership for customer invoices and restore previous
         membership state for customer refunds. Harmless on supplier ones.
-        We detect dynamically if the module account_invoice_refund_link is
-        installed for accurate source invoice.
         """
         self.mapped('invoice_line_ids.membership_lines').write({
             'state': 'canceled',
         })
         refunds = self.filtered(lambda r: (
-            r.type == 'out_refund' and
-            (r.origin or getattr(r, 'origin_invoice_ids', False))
+            r.type == 'out_refund' and r.refund_invoice_id
         ))
         for refund in refunds:
-            # Search for the original invoice it modifies
-            if hasattr(refund, 'origin_invoice_ids'):  # pragma: no cover
-                # If account_invoice_refund_link module is installed
-                origins = refund.origin_invoices_ids
-            else:
-                # Try to match by origin string
-                origins = self.search([
-                    ('type', '=', 'out_invoice'),
-                    ('number', '=', refund.origin),
-                ])
-            for origin in origins:
-                lines = origin.mapped('invoice_line_ids.membership_lines')
-                if origin and lines:
-                    origin_state = (
-                        'paid' if origin.state == 'paid' else 'invoiced'
-                    )
-                    lines.filtered(lambda r: r.state == 'canceled').write({
-                        'state': origin_state,
-                    })
-                    lines.write({
-                        'date_cancel': False,
-                    })
+            origin = refund.refund_invoice_id
```

Comment:
why do assign it to a variable?

Reply:
For shortening the reference that is later used.

Reply:
Ok, guessed it was something like that :smile: